Add  memory accounting for multi-phase execution (#15968)
crate/crate: docs/appendices/release-notes/5.7.2.rst

Fixes
=====

> - Added memory accounting for multi-phase execution to prevent that a node is
>   running out-of-memory caused by in-sub-queries such as
>   ''SELECT * FROM t1 where id in (SELECT id FROM t2)'' or lookup-joins with
>   large intermediate results.

Reviewer: ```suggestion
- Added memory accounting for multi-phase execution to prevent out-of-memory
  errors caused by sub-queries such as ''SELECT * FROM t1 WHERE id IN
  (SELECT id FROM t2)'' or lookup-joins with large intermediate results.
```
